src/System.CommandLine.Subsystems/Subsystems/CliSubsystem.cs

```diff
@@ -28,74 +29,26 @@ protected CliSubsystem(string name, SubsystemKind subsystemKind, IAnnotationProv
-    /// strongly typed explicit methods, such as help having `GetDescription(Symbol symbol)` method.
+    /// This value is protected because it is a convenience for subsystem authors. It calls <see cref="SymbolAnnotationStorageExtensions"/>
     /// </remarks>
     protected internal bool TryGetAnnotation<TValue>(CliSymbol symbol, AnnotationId<TValue> id, [NotNullWhen(true)] out TValue? value)
     {
-        if (_defaultProvider is not null && _defaultProvider.TryGet(symbol, id, out value))
-        {
-            return true;
-        }
-        if (_annotationProvider is not null && _annotationProvider.TryGet(symbol, id, out value))
-        {
-            return true;
-        }
-        value = default;
-        return false;
-    }
-
-    /// <inheritdoc cref="CliSubsystem.SetAnnotation{TValue}(CliSymbol, AnnotationId{TValue}, TValue)"/>
-    /// <returns>A delegate that returns the value.</returns>
-    protected internal bool TryGetAnnotation<TValue>(CliSymbol symbol, AnnotationId<Func<TValue>> id, [NotNullWhen(true)] out Func<TValue>? value)
-    {
-        if (_defaultProvider is not null && _defaultProvider.TryGet(symbol, id, out var storedValue))
-        {
-            value = storedValue;
-            return true;
-        }
-        if (_annotationProvider is not null && _annotationProvider.TryGet(symbol, id, out var storedValue2))
-        {
-            value = storedValue2;
-            return true;
-        }
-        value = default;
-        return false;
-    }
-
-    /// <summary>
-    /// Set the value for the symbol and annotation ID in the annotation provider.
-    /// </summary>
-    /// <typeparam name="TValue">The value of the type to set</typeparam>
-    /// <param name="symbol">The symbol the value is attached to</param>
-    /// <param name="id">The id for the value to be set. For example, an annotation ID for help is description</param>
-    /// <param name="value">An out parameter to contain the result</param>
-    /// <remarks>
-    /// This value is protected because these values are always retrieved from derived classes that offer
-    /// strongly typed explicit methods, such as help having `GetDescription(Symbol symbol, "My help description")` method.
-    /// </remarks>
-    protected internal void SetAnnotation<TValue>(CliSymbol symbol, AnnotationId<TValue> id, TValue value)
-    {
-        (_defaultProvider ??= new DefaultAnnotationProvider()).Set(symbol, id, value);
-    }
-
-    /// <inheritdoc cref="CliSubsystem.SetAnnotation{TValue}(CliSymbol, AnnotationId{TValue}, TValue)"/>
-    /// <param name="value">A delegate that returns the value.</param>
-    protected internal void SetAnnotation<TValue>(CliSymbol symbol, AnnotationId<Func<TValue>> id, Func<TValue> value)
-    {
-        (_defaultProvider ??= new DefaultAnnotationProvider()).Set<Func<TValue>>(symbol, id, value);
+        return symbol.TryGetAnnotation(id, out value, _annotationProvider);
```

Comment:
I'm not sure we still need this method. As a convenience, perhaps wait until we understand the calling code better.

Reply:
I will revise the comment. It's not just a convenience, it's the way that subsystem authors must access annotations if they want to respect the subsystem's `IAnnotationProvider`, as the provider is private.